distribution/pull_v2.go

```diff
@@ -877,6 +881,30 @@ func (p *v2Puller) pullSchema2Config(ctx context.Context, dgst digest.Digest) (c
 	return configJSON, nil
 }
 
+func retry(ctx context.Context, attempts int, sleep time.Duration, f func(ctx context.Context) error) (err error) {
+	for i := 0; i < attempts; i++ {
+		err = retryOnError(f(ctx))
+		if err == nil {
+			return nil
+		}
+		if _, isDNR := err.(xfer.DoNotRetry); isDNR {
+			logrus.WithError(err).Errorf("download failed after %d attempts", attempts)
+			return err
+		}
+
+		timer := time.NewTimer(sleep)
```

Comment:
Just a quick blurb (partially "note to self"); I recall there were some quirks with using timer, and I forgot exactly which one was better for this I see `xfer` uses a `ticker` instead, so wondering which one is best to use for this ( (https://gobyexample.com/tickers may indicate `ticker` ?); https://github.com/moby/moby/blob/65b8bcc3216ec64c8f761f35c6caba109b788254/distribution/xfer/download.go#L293